src/main/java/net/openhft/chronicle/network/internal/lookuptable/FileBasedHostnamePortLookupTable.java

```diff
@@ -232,15 +236,19 @@ private <T> T lockFileAndDo(Supplier<T> supplier, boolean shared) {
             final String message = "Failed to acquire lock on the shared mappings file. Retrying, file=" + sharedTableFile + ", count=" + count + ", elapsed=" + elapsedMs + " ms";
             Jvm.debug().on(FileBasedHostnamePortLookupTable.class, message, lastThrown);
         }
-        RuntimeException re = new RuntimeException("Couldn't acquire lock on shared mapping file " + sharedTableFile);
-        re.initCause(lastThrown);
-        throw re;
+        throw new RuntimeException("Couldn't acquire lock on shared mapping file " + sharedTableFile, lastThrown);
     }
 
     @Override
-    public synchronized void close() throws IOException {
+    protected synchronized void performClose() {
+        lockFileAndDo(() -> {
+            allMappings.clear();
+            sharedTableWire.clear();
+            zeroOutRemainingBytes(0);
+        }, false);
```

Comment:
On Windows the delete a few lines down won't work, but if we clear the file first it should have the same effect (i.e. clearing the registry)